### PR TITLE
Release MultiChain 0.5

### PR DESCRIPTION
**Pull request by darkwing**
SCR-20230525-jgwm

## Explanation

This pull request represents the release of the Multichain 0.5 design work.  This includes:

* A completely new application header, which contains:
   1.  A new network picker, which displays as only an avatar in the popup and as a full dropdown in full screen mode
   2. A new account picker
   3. A new connected icon which displays in popup mode
   4. A new global menu which contains controls that were formally in the account menu as well as account options menu
* A new token list
* A new token details popover

This PR is meant to achieve the following:

* Removes the `process.env.MULTICHAIN` flag and usage, exposing the new UI
* Removes legacy UI components, like the old network picker and account dropdown, that are no longer in use.
* Updates dozens of E2Es, since this core functionality and navigation paths have changed
* Removes legacy localization strings.

This PR is obviously large.  I've done my best to make each commit and its message meaningful so the reviewer can review by commit if they so choose.

## Pre-merge author checklist

- [ ] I've clearly explained:
  - [ ] What problem this PR is solving
  - [ ] How this problem was solved
  - [ ] How reviewers can test my changes
- [ ] Sufficient automated test coverage has been added

## Pre-merge reviewer checklist

- [ ] Manual testing (e.g. pull and build branch, run in browser, test code being changed)
- [ ] PR is linked to the appropriate GitHub issue
- [ ] **IF** this PR fixes a bug in the release milestone, add this PR to the release milestone

If further QA is required (e.g. new feature, complex testing steps, large refactor), add the `Extension QA Board` label.

In this case, a QA Engineer approval will be be required.

### CODE REVIEW

**Comment by DONOVA28** [comment]
Are we sure we are in the right meta-mask account because they didn't have two previous deleted MetaMask

**Review by zone-live — DISMISSED** [review]
most touched files are locales and e2e tests, UI wise looks good 👍🏼

**Comment by NidhiKJha** [comment]
Amazing work! A couple of suggestions:

In the locked state, the header needs to be fixed in popup
Screenshot 2023-05-31 at 6 02 35 PM

**Comment by NidhiKJha** [comment]
Question: Are we disabling auto suggestion for new account names?
https://github.com/MetaMask/metamask-extension/assets/39872794/28293433-35ad-478c-b653-db4ce013c1ad

**Comment by darkwing** [comment]
> Question: Are we disabling auto suggestion for new account names?

That's currently also an issue on develop, so not related to this PR.

**Review by vthomas13 — DISMISSED** [review]

**Comment by pedronfigueiredo** [comment]
This looks amazing David, can't wait to see it in production!

A few questions for my understanding:

1. When enabling test networks, the scroll state jumps to the bottom of the list hiding Mainnet which is currently selected. Would it be preferable to start that expanded view from the top?

2. There is a subtle but noticeable backdrop flicker when toggling on and off. Is this a function of the components we are relying on?

Screenshot 2023-06-01 at 12 07 05

3. Is there a way in which the UI could change somehow on click, indicating a loading state on the "Refresh list" button so that the user can "feel" it's working?

Screenshot 2023-06-01 at 12 24 09

**Comment by darkwing** [comment]
> 1. When enabling test networks, the scroll state jumps to the bottom of the list hiding Mainnet which is currently selected. 
 Would it be preferable to start that expanded view from the top?
 
Since the test networks display below the scrolling line, we decided to scroll down to them, as the user had the intent to see test networks

> 2. There is a subtle but noticeable backdrop flicker when toggling on and off. Is this a function of the components we are relying on?

I can look into it but hoping it doesn't block merge

> 3.  Is there a way in which the UI could change somehow on click, indicating a loading state on the "Refresh list" button so that the user can "feel" it's working?

Good idea we can follow up on -- we currently have this same functionality without response, so it shouldn't block.

File: ui/components/ui/list-item/index.scss
Comment (NidhiKJha) [inline ui/components/ui/list-item/index.scss:10]: We removed the showBorder condition here and that resulted in the border top in activity list. I think we can remove it permanently now from css. Since we are removing the feature flag?
Screenshot 2023-06-01 at 9 04 23 PM


```suggestion

```

**Review by NidhiKJha — CHANGES_REQUESTED** [review]
Everything looks great both feature wise and UI wise. Added a small UI change and also I am still seeing the double fox icon
Screenshot 2023-06-01 at 9 02 18 PM

**Comment by darkwing** [comment]
I cannot duplicate the double logos, can you make sure you have the most up to date PR code?

**Review by NidhiKJha — APPROVED** [review]
Excellent!! 😍

**Comment by pedronfigueiredo** [comment]
@darkwing, thanks for your comments. I didn't mean any of my questions above as blockers.

I am tracking them here https://github.com/MetaMask/metamask-extension/issues/19392 and https://github.com/MetaMask/metamask-extension/issues/19393 as separate tasks for us to consider at another time.

Can you and @kevinghim find a good epic for them?

**Review by pedronfigueiredo — APPROVED** [review]

**Review by kumavis — APPROVED** [review]
:+1:  approving code changes only in capacity to my codeowners reference (trivial change under `development` dir)

great step forward toward the multichain! thanks to all who pushed this forward :tada:

**Review by Mrtenz — APPROVED** [review]
The changes related to Snaps look good to me.

**Pull request merged**
